src/conn/conn_api.c

```diff
@@ -1872,7 +1877,10 @@ wiredtiger_open(const char *home, WT_EVENT_HANDLER *event_handler,
 	    WIREDTIGER_VERSION_MAJOR, WIREDTIGER_VERSION_MINOR) >=
 	    (int)sizeof(version), ENOMEM);
 	__conn_config_append(cfg, version);
-	WT_ERR(__conn_config_file(session, WT_BASECONFIG, 0, cfg, i1));
+
+	/* Ignore the base_config file if we config_base set to false. */
+	if (config_base_set)
```

Comment:
Our style is to only do `if (var)` for simple `int` vars.  So either make `config_base_set` an `int`, and set it above with `config_base_set = (cval.val != 0);` or just change this to `if (config_base_set != 0)`.